temporal-test-server/src/main/java/io/temporal/internal/testservice/TestWorkflowMutableStateImpl.java

```diff
@@ -3502,26 +3526,26 @@ private void addExecutionSignaledByExternalEvent(
 
   private void addWorkflowExecutionOptionsUpdatedEvent(
       RequestContext ctx, String requestId, List<Callback> completionCallbacks, List<Link> links) {
+    HistoryEvent.Builder event =
+        HistoryEvent.newBuilder()
+            .setWorkerMayIgnore(true)
+            .setEventType(EVENT_TYPE_WORKFLOW_EXECUTION_OPTIONS_UPDATED);
+
     WorkflowExecutionOptionsUpdatedEventAttributes.Builder attrs =
         WorkflowExecutionOptionsUpdatedEventAttributes.newBuilder();
     if (requestId != null) {
       attrs.setAttachedRequestId(requestId);
-      this.attachedRequestIds.add(requestId);
+      this.attachRequestId(requestId, event.getEventType(), BUFFERED_EVENT_ID);
```

Comment:
Is this necessarily buffered?

Reply:
Unsure, but it doesn't really matter. In the test server, when you add the event to the context, it gets an event id, but it doesn't seem to necessarily the final event id (which is weird, but make sense since it might have the workflow task completed added before flushing the context).
I can assume that it is buffered here, and if the context flushes the events, it will assign the final event id at that point, and this will be updated.